File: Chapitre3/exercice3.2.py
import os
import subprocess
import sys

def main():
    home = os.environ.get("HOME")
    if home is None:
        return 1

    nouveau_repertoire = os.path.join(home, "nouveau_repertoire")

    subprocess.run(["mkdir", nouveau_repertoire])

    os.execv("/usr/bin/cp", ["cp", "-r", ".", nouveau_repertoire])

    print("Erreur execv", file=sys.stderr)
    return 1

if __name__ == "__main__":
    sys.exit(main())

File: Chapitre3/exercice3.2.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>

int main() {
    char *home = getenv("HOME");
    char nouveau_repertoire;
    char commande_mkdir;

    if (home == NULL) return 1;

    sprintf(nouveau_repertoire, "%s/nouveau_repertoire", home);
    
    sprintf(commande_mkdir, "mkdir %s", nouveau_repertoire);
    system(commande_mkdir);

    char *argv[] = {"cp", "-r", ".", nouveau_repertoire, NULL};
    execv("/usr/bin/cp", argv);
    
    perror("Erreur execv");
    return 1;
}
